janusgraph-dist/README.md

```diff
@@ -68,6 +68,178 @@ To build the .deb and .rpm packages:
 To delete the packaging scripts from the root of the repository, run
 `mvn -N -Ppkg-tools clean` from the janusgraph-dist module.
 
+Building the Docker Image
+-------------------------
+
+Requirements
+
+* [Docker](https://www.docker.com/)
+
+First, build the zip archive from the previous section. Next run the command
+`mvn dockerfile:build -Pjanusgraph-docker -pl janusgraph-dist` from the root directory of this 
+repository. This command will build a JanusGraph docker image tagged with the current project 
+version. If you want to specify a different tag, you can run the command as
+`mvn dockerfile:build -Ddocker.tag=latest -Pjanusgraph-docker -pl janusgraph-dist`.
+
+Running the Docker Image
+------------------------
+
+By default, the docker image is built using JanusGraph's [in memory](http://docs.janusgraph.org/latest/inmemorystorage.html)
```

Comment:
I think the default should be Cassandra+Elasticsearch

Reply:
@sjudeng I'm working on taking the implementation I moved outside of this repository to incorporate it in this PR. 

Here are my thoughts on why I had the initial implementation as in memory although I'm actually planning on changing it to be berkleydb + lucene.

Docker images are supposed to start a single process only per their best practices. https://docs.docker.com/develop/develop-images/dockerfile_best-practices/. We could use cassandra embedded but we would still need an elastic search instance to be running.

When I consume other database implementations docker images, I'm normally using them as some sort of integration layer for testing my applications against them. In that context, I normally just desire that the interaction with the database behaves as expected rather than it being representative of a potential production deployment. I'm looking for speed rather than a true end to end representation. I think this also rings true for anyone who is just looking to "play" with JanusGraph.

I also might use those docker images to persist data during my day to day application development without having to have an actual implementation installed on my machine. For instance, I have one docker compose file that I use for my fast integration tests (in memory) and another that I use for my typical development where I want to interact with data I've accumulated during development that persists between restarts (berkley db + lucene). We can accomplish the persistence by mounting a directory as a data volume that will store the data between restarts of the containers themselves.

Example:

```yml
version: "3"

services:
  janusgraph:
    image: experoinc/janusgraph:0.2.0
    container_name: janusgraph-mount
    ports:
      - "8182:8182"
    volumes:
      # named volume mounts
      - "janusgraph-mount-data:/var/lib/janusgraph"
      # bind mounts for configs; use read only so not overridden by environment variables
      - "./example-config:/etc/opt/janusgraph/:ro"

# use a named volume to maintain database data between restarts
volumes:
  janusgraph-mount-data:
```

We can provide docker-compose files that represent a more complex deployment using Cassandra and Elasticsearch but I don't think that should be the initial implementation of the image.

https://github.com/experoinc/janusgraph-images/blob/master/docker-compose-cassandra-es.yml

Let me know what you think.

Reply:
@chrislbs Thanks for the explanation. You're right the image should be functional without additional (Cassandra, Elasticsearch, etc.) containers. Only other thing would be supporting data persistence but I think you're on the right track with berkleydb+lucene.